app/services/update_namespace_stats.rb

```diff
@@ -0,0 +1,88 @@
+# frozen_string_literal: true
+
+class UpdateNamespaceStats
+
+  def initialize(course)
+    @course = course
+    update_stats
+  end
+
+  def update_stats
+    wiki_namespaces = @course.tracked_namespaces
+    stats = {}
+
+    wiki_namespaces.each do |wiki_ns|
+      @wiki = wiki(wiki_ns[:wiki])
+      namespaces = wiki_ns[:namespaces]
+      wiki_stats = {}
+      namespaces.each do |ns|
+        @namespace = ns
+        wiki_stats[@namespace] = {
+          'edited_count': edited_articles_count,
+          'new_count': new_articles_count,
+          'revision_count': revision_count,
+          'user_count': user_count,
+          'word_count': word_count,
+          'references_count': references_count,
+          'views_count': views_count
+        }
+      end
+      stats[@wiki.domain] = wiki_stats
+    end
+    
+    crs_stat = CourseStat.find_by(course_id: @course.id) || CourseStat.create(course_id: @course.id)
```

Comment:
I prefer avoiding abbreviations like this in variable names. Better to use a longer name and spell out the words more explicitly, to make it easier to read and figure out what's going on.